datahost-ld-openapi/build.clj

```diff
@@ -23,30 +24,31 @@
                    "branch --show-current"]
                   (map #(tag (b/git-process {:git-args %})))
                   (remove nil?))
-        image-type (get opts :image-type :docker)]
-    (pack/docker
-     {:basis (b/create-basis {:project "deps.edn" :aliases [:ldapi/docker]})
-      ;; If we don't include a tag in the :image-name, then pack implicitly
-      ;; tags the image with latest, even when we specify additional tags. So
-      ;; choose a tag arbitrarily to be part of the :image-name, and then
-      ;; provide the rest in :tags.
-      :image-name (str repo "/datahost-ld-openapi:" (first tags))
-      :tags (set (rest tags))
-      :image-type image-type
-
-      :base-image "eclipse-temurin:17" ;; An openJDK 17 base docker provided by https://github.com/adoptium/containers#containers
-
-      :platforms #{:linux/amd64 :linux/arm64}
-
-      ;; NOTE Not as documented!
-      ;; The docstring states that these should be
-      ;;     :to-registry {:username ... :password ...}
-      ;; but alas, that is a lie.
-      ;; https://github.com/juxt/pack.alpha/issues/101
-      :to-registry-username "_json_key"
-      :to-registry-password (System/getenv "GCLOUD_SERVICE_KEY")
-
-      })))
+        image-type (get opts :image-type :docker)
+        ^JibContainer container (pack/docker
+                                  {:basis (b/create-basis {:project "deps.edn" :aliases [:ldapi/docker]})
+                                   ;; If we don't include a tag in the :image-name, then pack implicitly
+                                   ;; tags the image with latest, even when we specify additional tags. So
+                                   ;; choose a tag arbitrarily to be part of the :image-name, and then
+                                   ;; provide the rest in :tags.
+                                   :image-name (str repo "/datahost-ld-openapi:" (first tags))
+                                   :tags (set (rest tags))
+                                   :image-type image-type
+
+                                   :base-image "eclipse-temurin:17" ;; An openJDK 17 base docker provided by https://github.com/adoptium/containers#containers
+
+                                   :platforms #{:linux/amd64 :linux/arm64}
+
+                                   ;; NOTE Not as documented!
+                                   ;; The docstring states that these should be
+                                   ;;     :to-registry {:username ... :password ...}
+                                   ;; but alas, that is a lie.
+                                   ;; https://github.com/juxt/pack.alpha/issues/101
+                                   :to-registry-username "_json_key"
+                                   :to-registry-password (System/getenv "GCLOUD_SERVICE_KEY")
+
+                                   })]
+    (println (.. container (getDigest) (getHash)))))
```

Comment:
is this important? or from debugging?

Reply:
Yes we need this to know which SHA of the application to deploy. The CI process writes it to file and then supplies it as the `digest` parameter to `terrafrom plan`.